core/src/main/java/software/amazon/awssdk/internal/net/ConnectionUtils.java

```diff
@@ -17,37 +17,30 @@
 
 import java.io.IOException;
 import java.net.HttpURLConnection;
+import java.net.Proxy;
 import java.net.URI;
+import java.util.Map;
 import software.amazon.awssdk.annotation.SdkInternalApi;
 
 @SdkInternalApi
 public class ConnectionUtils {
 
-    private static volatile ConnectionUtils instance;
+    private static final ConnectionUtils INSTANCE = new ConnectionUtils();
```

Comment:
Would it be better if we just made all of this static instead of a singleton, or are we mocking it out for testing?

Reply:
This is mocked out when testing `HttpCredentialsUtils`